### PR TITLE
VideoPress: Support `size`, `label` and `type` at `Input` component

### PR DESCRIPTION
**Pull request by renatoagds**
Open support for 3 new props at `Input` component: `size`, `label` and `type`.

Related with #26036 

#### Changes proposed in this Pull Request:

  * Add support for `type`, `size` and `label` props at `Input`.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start `Storybook`
* Navigate to `VideoPress/Components/Input`
* Interact with new stories and controls.

#### Demo

https://user-images.githubusercontent.com/1663717/189215019-58b0f4e7-34b6-4f1f-90db-60b4150b94ff.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/components/input/types.ts [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
+} & InputBaseProps;
 
+type TextArea = React.TextareaHTMLAttributes< HTMLTextAreaElement > & {
 	/**
-	 * Callback to be invoked when the user presses the Enter key.
+	 * No support for icon when using textarea.
 	 */
-	onEnter: ( value: string ) => unknown;
-};
+	icon?: undefined;
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/admin/components/input/types.ts:45]: Maybe it is better to not declare `icon`?

File: projects/packages/videopress/src/client/admin/components/input/types.ts [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
+	/**
+	 * Only use these types if is textarea.
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/admin/components/input/types.ts]:
```suggestion
	 * Only available type for TextArea type.
```

File: projects/packages/videopress/src/client/admin/components/input/types.ts (hunk 1 above)
Comment (renatoagds) [reply projects/packages/videopress/src/client/admin/components/input/types.ts:45]: Nope. We need to declare as undefined to Typescript understand that this property doesn't exist at this moment

File: projects/packages/videopress/src/client/admin/components/input/types.ts (hunk 2 above)
Comment (renatoagds) [reply projects/packages/videopress/src/client/admin/components/input/types.ts]: I don't think this is exactly the description that I want to use 🤔  ... It Is more a informative comment saying that we fix `type: 'textarea'` to make typescript understand which type it needs to use

File: projects/packages/videopress/src/client/admin/components/input/types.ts (hunk 1 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/types.ts:45]: Ah, I see now in the `InputProps`.

File: projects/packages/videopress/src/client/admin/components/input/types.ts (hunk 2 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/types.ts]: Maybe something like `Fixed as "textarea" when the input is a textarea element`.

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 3]
```diff
@@ -37,27 +31,65 @@ export const Input = ( {
 	);
 
 	const handleKeyUpEvent = useCallback(
-		( e: KeyboardEvent< HTMLInputElement > ) => {
+		( e: KeyboardEvent< HTMLInputElement | HTMLTextAreaElement > ) => {
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx:34]: Something that I just noticed, `e.key` is `Enter` both on normal and numpad enter keys, but `e.code` can be `Enter` or `NumpadEnter`, so we need something like `if ( onEnter != null && [ 'Enter', 'NumpadEnter' ].includes( e.code ) ) {`

Comment (renatoagds) [reply projects/packages/videopress/src/client/admin/components/input/index.tsx:34]: We could jump to `e.key` if we want. But can we open an issue and map this?

Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/index.tsx:34]: Sure :+1:

**Review by dhasilva — APPROVED** [review]
Noice

File: projects/packages/videopress/src/client/admin/components/input/types.ts (hunk 2 above)
Comment (renatoagds) [reply projects/packages/videopress/src/client/admin/components/input/types.ts]: @dhasilva tried to improve the comment, but I don't think we have too much to add in this case, it should be understandable for those who know TS.

File: projects/packages/videopress/src/client/admin/components/input/index.tsx (hunk 3 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/index.tsx:34]: https://github.com/Automattic/jetpack/issues/26138

**Pull request merged**